src/random.rs

```diff
@@ -0,0 +1,136 @@
+use core::{marker::PhantomData, mem::size_of};
+
+use embedded_hal::blocking::rng::Read;
+
+use rand::{RngCore, SeedableRng};
+
+use crate::hwrng::HWRNG;
+
+/// Wrapper around RIOTs `random` module.
+///
+/// ## Seedlength
+/// Since the module allows a dynamic seedsize
+/// it needs to be specified in the type.
+/// The `SEED_LENGTH` variable specifies the seedlength in bytes.
+/// **Since RIOT takes in `uint32_t` (`u32`) the length need to be divisable by 4!**
+///
+/// ## Security
+/// Even though `Random` claims to be a cryptographic secure prng
+/// it only can be if provided sufficently random seeds! See remarks at [`crate::hwrng::HWRNG`]
+/// if when using it to generate seeds.
+///
+/// ## Global state
+/// Be aware that there should be only one `Random` object at a time,
+/// since RIOT uses a global state for this internally, so creating a second object
+/// just results in the global state beeing overwritten and
+/// both objects representing practically the same prng.
+#[derive(Debug)]
+pub struct Random<const SEED_LENGTH: usize> {
+    // Make sure this gets not manually constructed
+    private: PhantomData<()>,
```

Comment:
```suggestion
    _private: (),
```

Unless the PhantomData should make this not Send/Sync (for which it would need to contain something that's not Send/Sync), it suffices to just have the unit type in here. (And an unused field should be underscore prefixed IIRC).

Reply:
Addressed in https://github.com/RIOT-OS/rust-riot-wrappers/pull/54/commits/2b99ee539048daabddab05cc8ff2a2e6acffa946